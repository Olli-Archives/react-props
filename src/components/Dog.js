import React from 'react';
import PropTypes from 'prop-types';

export default function Dog({ name, age, weight }){
  return (
    <ul>
      <li>name:{name}</li>
      <li>age:{age}</li>
      <li>weight:{weight}</li>
    </ul>
  );
}

Dog.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  weight: PropTypes.string
}
;