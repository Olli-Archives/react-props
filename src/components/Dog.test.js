import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('dog component', ()=>{
  it('renders dog', ()=>{
    const wrapper = shallow(<Dog name='skip' age={6} weight='700'/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
