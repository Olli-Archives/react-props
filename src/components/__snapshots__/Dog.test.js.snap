// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dog component renders dog 1`] = `
<ul>
  <li>
    name:
    skip
  </li>
  <li>
    age:
    6
  </li>
  <li>
    weight:
    700
  </li>
</ul>
`;
